### PR TITLE
Py3 fixes for shmemutils

### PR DESCRIPTION
**Pull request by klaussfreire**
- Avoid buffer uses
- Be careful to release all buffers before trying to close the mmap
- bytes vs strings

Tests are part of PR #27

### CODE REVIEW

File: lib/chorde/shmemutils.py [hunk 1]
```diff
@@ -352,15 +352,26 @@ def close(self):
         if bitmap is not None:
             # Release slot
             self.bitmap_slot.value = False
+        del bitmap
```
Comment (chpedersen) [inline lib/chorde/shmemutils.py:355]: Is this necessary?

Comment (klaussfreire) [reply lib/chorde/shmemutils.py:355]: Yes, we can't still have references to the bitmap when we .close() it

**Comment by chpedersen** [comment]
Line 159:

```python3
                for offs in range(slots):
                    nslot = (slot+offs) % slots
                    if not bitmap[nslot]:
                        slot = nslot
                else:
                    raise AssertionError("All slots occupied")
```
shouldn't there be a break after `slot = nslot` ?

**Comment by klaussfreire** [comment]
> Line 159:
> 
> ```python
>                 for offs in range(slots):
>                     nslot = (slot+offs) % slots
>                     if not bitmap[nslot]:
>                         slot = nslot
>                 else:
>                     raise AssertionError("All slots occupied")
> ```
> 
> shouldn't there be a break after `slot = nslot` ?

Indeed, but it's not a py3 bug. I put the fix in PR #32 

It seems collisions don't happen often, or we would have noticed earlier.

**Review by chpedersen — APPROVED** [review]

**Review by mauguignard — APPROVED** [review]

**Pull request merged**